[DRAFT] Che docs contribution (#59)
eclipse-che/blog: _posts/2024-08-01-contributing-to-che-docs.adoc

---
title: Contributing to Eclipse Che documentation
layout: post
author: Jana Vrbkova
description: >-
  Contribute to Eclipse Che documentation using an Eclipse Che workspace.
categories: []
keywords: ['che-docs', 'documentation', 'write', 'contribute']
slug: /@deerskindoll/contributing-to-che-docs
---

= Contributing to Eclipse Che documentation

Writing about Eclipse Che is even easier than you think.
The link:https://github.com/eclipse-che/che-docs[Eclipse Che documentation repository on GitHub] was created with community contribution in mind,
featuring tools you need to create and test your content.
And from the initial draft to submitting a PR,
you can do it all in the environment you're already familiar with: an Eclipse Che workspace.

.Getting started

> Use either link:https://eclipse.dev/che/docs/stable/administration-guide/installing-che/[your own Eclipse Che installation]

Reviewer: Let's mention it as a second option, or not mention it at all

Author: got it

or click the Developer Workspace button in the che-docs repository README file.
This opens a clone of the repository in an Eclipse Che workspace hosted

> by Red Hat in the link:https://developers.redhat.com/developer-sandbox?source=sso[Developer Sandbox].

Reviewer: ```suggestion
by Red Hat in the link:https://developers.redhat.com/developer-sandbox[Developer Sandbox].
```

Alternatively, you can click this button:

> [![Contribute](https://www.eclipse.org/che/contribute.svg)](https://workspaces.openshift.com/f?url=https://github.com/eclipse/che-docs)

Reviewer: ```suggestion
[![Contribute](https://www.eclipse.org/che/contribute.svg)](https://workspaces.openshift.com#https://github.com/eclipse/che-docs)
```

---

> title: Contributing to Eclipse Che documentation

Reviewer: let's make the title catchy

> title: Contributing to Eclipse Che documentation

Reviewer: ```suggestion
title: Contributing to Eclipse Che documentation with one click
```

Reviewer: not ideal, so feel free to come up with smth. better but as a reader, we should have smth. in the title that would grab the attention (the current one is very neutral for the blog imo)

layout: post
author: Jana Vrbkova
description: >-

>   Contribute to Eclipse Che documentation using an Eclipse Che workspace.

Reviewer 2: ```suggestion
  Contribute to Eclipse Che documentation using an Eclipse Che cloud development environment.
```

Author: if I understand it correctly, it's better to refer to it as a CDE than a workspace (in general)? I want to make sure I use the correct expression in the future.

categories: []
keywords: ['che-docs', 'documentation', 'write', 'contribute']
slug: /@deerskindoll/contributing-to-che-docs
---

= Contributing to Eclipse Che documentation

Writing about Eclipse Che is even easier than you think.
The link:https://github.com/eclipse-che/che-docs[Eclipse Che documentation repository on GitHub] was created with community contribution in mind,
featuring tools you need to create and test your content.
And from the initial draft to submitting a PR,

> you can do it all in the environment you're already familiar with: an Eclipse Che workspace.

Reviewer 2: ```suggestion
you can do it all in the environment you're already familiar with: an Eclipse Che cloud development environment (CDE).
```

> .Getting started

Reviewer: Please add `Devfile and How it Works?` section or smth. similar that would explain CDE, predefined commands etc.

Reviewer: image that might be useful here 
<img width="1137" alt="Screenshot 2024-07-26 at 12 27 07" src="https://github.com/user-attachments/assets/4b561bb4-d22d-4a28-bc15-c5833b540506">

Use either link:https://eclipse.dev/che/docs/stable/administration-guide/installing-che/[your already installed Eclipse Che]
or click the Developer Workspace button directly in the che-docs repository

> to directly open a clone of the repository in an Eclipse Che workspace hosted

Reviewer 2: ```suggestion
to directly open a clone of the repository in an Eclipse Che CDE hosted
```

> by Red Hat in the the link:https://developers.redhat.com/developer-sandbox?source=sso[Developer Sandbox].

Reviewer 2: ```suggestion
by Red Hat in the link:https://developers.redhat.com/developer-sandbox?source=sso[Developer Sandbox].
```

> *add the button?*

Reviewer: yes

> *describe plugins?*

Reviewer: yes

> *vale sync command*

Reviewer: yes

The Vale extension included in the che-docs repository follows the grammatical,
spelling and stylistic rules from the link:https://redhat-documentation.github.io/supplementary-style-guide/[Red Hat supplementary style guide].
Before you start writing,
read the section about link:https://redhat-documentation.github.io/supplementary-style-guide/#technical-examples[documenting commands].

> .Writing tips

Reviewer: really like this section - a good way to wrap the blog up 👍

> A good technical document is easy to read and easy to follow.

Reviewer: ```suggestion
A good technical document is easy to read, easy to follow, and now easy to contribute!
```

Reviewer: not ideal suggestion (feel free to skip it) but somewhere in the article we should articulate that the game-changer is contributing and spinning up an environment for contribution with one click

Here are a few tips that will help you write great technical content:

* Keep it simple. Avoid self-referential language ("this article is about..."), overly long introductions, and deep-dives into the feature's history.
* Keep it accessible. Short paragraphs, sentences and words make your content easy to read.

> * Keep the focus on the user. Write the content around what the user can achieve when they follow your instructions.

Reviewer 2: If it makes sense here, I think a bullet point about avoiding passive voice would be helpful

Author: added
